src/heuristic/HeuristicMapper.cpp

```diff
@@ -438,45 +507,48 @@ void HeuristicMapper::expandNode(
 
   // set up new teleportation qubits
   std::set<Edge> perms = architecture.getCouplingMap();
-  architecture.getCurrentTeleportations().clear();
-  architecture.getTeleportationQubits().clear();
-  for (std::size_t i = 0; i < results.config.teleportationQubits; i += 2) {
-    architecture.getTeleportationQubits().emplace_back(
-        node.locations.at(qc.getNqubits() + i),
-        node.locations.at(qc.getNqubits() + i + 1));
-    Edge e;
-    for (auto const& g : architecture.getCouplingMap()) {
-      if (g.first == node.locations.at(qc.getNqubits() + i) &&
-          g.second != node.locations.at(qc.getNqubits() + i + 1)) {
-        e.first  = g.second;
-        e.second = static_cast<std::uint16_t>(
-            node.locations.at(qc.getNqubits() + i + 1));
-        architecture.getCurrentTeleportations().insert(e);
-        perms.insert(e);
-      }
-      if (g.second == node.locations.at(qc.getNqubits() + i) &&
-          g.first != node.locations.at(qc.getNqubits() + i + 1)) {
-        e.first  = g.first;
-        e.second = static_cast<std::uint16_t>(
-            node.locations.at(qc.getNqubits() + i + 1));
-        architecture.getCurrentTeleportations().insert(e);
-        perms.insert(e);
-      }
-      if (g.first == node.locations.at(qc.getNqubits() + i + 1) &&
-          g.second != node.locations.at(qc.getNqubits() + i)) {
-        e.first = g.second;
-        e.second =
-            static_cast<std::uint16_t>(node.locations.at(qc.getNqubits() + i));
-        architecture.getCurrentTeleportations().insert(e);
-        perms.insert(e);
-      }
-      if (g.second == node.locations.at(qc.getNqubits() + i + 1) &&
-          g.first != node.locations.at(qc.getNqubits() + i)) {
-        e.first = g.first;
-        e.second =
-            static_cast<std::uint16_t>(node.locations.at(qc.getNqubits() + i));
-        architecture.getCurrentTeleportations().insert(e);
-        perms.insert(e);
+  // for now teleportation is not supported for noise-aware mapping
+  if (!results.config.considerFidelity) {
```

Comment:
Instead of wrapping everything in an if and increasing the indentation level, prefer to just return early for the not-supported case.

Reply:
Returning early would not result in the same behavior here, as I just wrapped the part of the method, which added teleportation qubits into `perms` (which admittedly is a large chunk of the function), but the execution flow would still continue below that with the usual mapping. However, I instead set `configuration.teleportationQubits = 0` now at the beginning of `HeuristicMapper::map` with an accompanying warning message, which is cleaner anyway.

Reply:
Ah, missed that on the phone and thought the function ended at that point. Sorry for the confusion.
Your change makes sense 👍🏼